src/Api/Tracking.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+declare(strict_types=1);
+
+namespace P4\MasterTheme\Api;
+
+use WP_REST_Server;
+
+/**
+ * Instance Tracking API
+ */
+class Tracking
+{
+    /**
+     * Register endpoint to read general Tracking.
+     *
+     * @example GET /wp-json/planet4/v1/tracking/
+     */
+    public static function register_endpoint(): void
+    {
+        register_rest_route(
+            'planet4/v1',
+            'tracking',
+            [
+                'methods' => WP_REST_Server::READABLE,
+                'callback' => function ($request) {
+                    $params = $request->get_params();
+
+                    return [
+                        'logins' => self::get_logins($params),
+                    ];
+                },
+                'permission_callback' => function ($request) {
+                    if (current_user_can('manage_options')) {
+                        return true;
+                    }
+
+                    if (!defined('PLANET4_API_KEY') || empty(PLANET4_API_KEY)) {
+                        return false;
+                    }
+
+                    $token = $request->get_header('X-Auth-Token');
+                    return !empty($token) && $token === PLANET4_API_KEY;
+                },
+                'args' => array(
+                    'last_days' => array(
+                        'required' => false,
+                        'default' => 30,
+                    ),
+                    'full' => array(
+                        'required' => false,
+                        'default' => false,
+                    ),
+                ),
+            ]
+        );
+    }
+
+    /**
+     * Get last logins filtered by days.
+     *
+     * @param array $params refers to request params
+     * @return array Get logins.
+    */
+    private static function get_logins(array $params): array
+    {
+        global $wpdb;
+
+        $response = [];
+        $last_days = (60 * 60 * 24 * $params['last_days']);
+        $now = time();
+
+        $query = $wpdb->get_results(
```

Comment:
We can improve this query as below -
```
$query = $wpdb->get_results(
    $wpdb->prepare(
        "SELECT u.display_name, m.meta_value
        FROM {$wpdb->usermeta} AS m
        INNER JOIN {$wpdb->users} AS u ON m.user_id = u.ID
        WHERE m.meta_key = %s",
        'session_tokens'
    ),
    OBJECT
);
```

- The query fetches all columns using (`SELECT *`), but only `display_name` and `meta_value` are needed. Using (`SELECT u.display_name, m.meta_value`) will improve efficiency.
- The table names (`wp_users, wp_usermeta`) are hardcoded, which can cause issues if the database prefix (wp_) changes.
- Using a prepare() query ensures that values are safely escaped, preventing SQL injection vulnerabilities.